dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityStore.java

```diff
@@ -1445,4 +1445,33 @@ private String getOrgUnitNameByUid(String uid) {
   private boolean skipOwnershipCheck(TrackedEntityQueryParams params) {
     return params.getUser() != null && params.getUser().isSuper();
   }
+
+  // TODO This has to be removed once SettingKey.TrackedEntityInstanceMaxLimit is removed
+  private int resolveTrackedEntityMaxLimit() {
+    int deprecatedTeiMaxLimit =
+        systemSettingManager.getIntegerSetting(SettingKey.DEPRECATED_TRACKED_ENTITY_MAX_LIMIT);
+    int newTeiMaxLimit =
+        systemSettingManager.getIntegerSetting(SettingKey.TRACKED_ENTITY_MAX_LIMIT);
+
+    if (isSet(deprecatedTeiMaxLimit)
+        && isSet(newTeiMaxLimit)
+        && deprecatedTeiMaxLimit != newTeiMaxLimit) {
+      throw new IllegalQueryException(
```

Comment:
@enricocolasante I think `IllegalStateException` seems more accurate to me. What do you think?